SU-HSDO/suhumsci
# STN-141 : External URL bug with incorrect URL path and missing icon

# [STN-141](https://sparkbox.atlassian.net/browse/STN-141) READY FOR REVIEW

## Summary
Original Bug: The horizontal card view `link to external` output incorrect information within in the HTML anchor tag. This is fixed by updating Drupal settings. Instructions are provided below.

Secondary Bug: Once the Drupal settings are updated, the icon beside the external link was hidden by `display: none`. This was resolved by surrounding the external link with a `div` and applying default classes which are referenced in our styles.

## Need Review By (Date)
ASAP

## Urgency
High

## Steps to Test
You can choose to test this PR by exporting a view or editing a view within your Drupal configuration.

### Test with an export
1. Run `npm run test` and confirm that all tests and linters pass.
2. Import the SBX News view: [SBX.txt](https://github.com/SU-HSDO/suhumsci/files/4150177/SBX.txt)
3. Create a basic page with a view with `View` set to `SBX News` and `Display` set to `Featured News Item`
    <img width="519" alt="basic page" src="https://user-images.githubusercontent.com/12678977/73686903-e5a42500-4696-11ea-830a-e901e57eaa8e.png">
4. Click `Save`
5. Under `Content` find and click on `Vertical card test page` to review the URL bug fixes
6. Click on the external link and confirm it redirects to seesparkbox.com
7. Confirm the external link displays the correct icon
8. Click on the link to content arrow and confirm it redirects to the `News Item Internal URL Bug Fix` page

### Test without an export
1. Run `npm run test` and confirm that all tests and linters pass.
2. Your horizontal card view settings should have the following:
    * In the Drupal admin panel, go to a horizontal card view and edit.
    * Under `Displays` click Show: `Settings`
    * <img width="370" alt="settings" src="https://user-images.githubusercontent.com/12678977/73687297-ae824380-4697-11ea-853d-74f3cfeae721.png">
    * Update the `Destination` for `Link to Content` to `URL`
    * Update the `Destination` for `Link to External` to `Location`
    * Click `Apply`
    * Now check your `Fields` settings
    * Click on `Content: Link to Content`
    * `Output the URL as text` should be checked
    * Click `Apply`
    * Click on `Content: Link to External Article`
    * `URL Only` *should not* be checked
    * Go to `Style Settings`
    * Check `Customize field and label wrapper HTML`
    * In the `Wrapper HTML element` dropdown, select `DIV`
    * Check `Add default classes`
    * Click `Apply`
    * Click `Save`
3. Under `Content` find and click on the page with the horizontal card view to review the URL bug fixes
4. Click on the external link and confirm it redirects to an external link
5. Confirm the external link displays the correct icon
6. Click on the link to content arrow and confirm it redirects to the correct internal content

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
